core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -31,6 +31,20 @@
   private TableScanUtil() {
   }
 
+  public static boolean hasDeletes(CombinedScanTask task) {
+    for (FileScanTask fileTask : task.files()) {
```

Comment:
Could be stream.anyMatch(TableScanUtil::hasDeletes) , but I think that's just a personal style decision.